File: AD-HOC/2293/main.c
#include <stdio.h>

void getField(int n, int m, int field[n][m]) {
    for (int i = 0; i < n; i++)
        for (int j = 0; j < m; j++)
            scanf("%d", &field[i][j]);
}

int getMaxWorms(int n, int m, int field[n][m]) {
    int maxWorms = 0;
    for (int i = 0; i < n; i++) {
        int worms = 0;
        for (int j = 0; j < m; j++)
            worms += field[i][j];
        if (worms > maxWorms) maxWorms = worms;
    }
    for (int j = 0; j < m; j++) {
        int worms = 0;
        for (int i = 0; i < n; i++)
            worms += field[i][j];
        if (worms > maxWorms) maxWorms = worms;
    }
    return maxWorms;
}

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    int field[n][m];
    getField(n, m, field);
    printf("%d\n", getMaxWorms(n, m, field));
    return 0;
}

File: AD-HOC/2293/main.js
function getField(n) {
    let field = [];
    while (n--)
        field.push(lines.shift().split(' ').map(Number));
    return field;
}

function getMaxWorms(n, m, field) {
    let maxWorms = 0;
    for (let i = 0; i < n; i++) {
        let worms = 0;
        for (let j = 0; j < m; j++)
            worms += field[i][j];
        if (worms > maxWorms) maxWorms = worms;
    }
    for (let j = 0; j < m; j++) {
        let worms = 0;
        for (let i = 0; i < n; i++)
            worms += field[i][j];
        if (worms > maxWorms) maxWorms = worms;
    }
    return maxWorms;
}

let lines = require('fs').readFileSync('/dev/stdin', 'utf-8').split('\n');
let [n, m] = lines.shift().split(' ').map(Number);
console.log(getMaxWorms(n, m, getField(n)));
